presentation/src/main/java/com/dcs/presentation/main/MainBottomNavigation.kt

```diff
@@ -0,0 +1,72 @@
+package com.dcs.presentation.main
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.material3.Icon
+import androidx.compose.material3.NavigationBar
+import androidx.compose.material3.NavigationBarItem
+import androidx.compose.material3.NavigationBarItemDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.navigation.NavGraph.Companion.findStartDestination
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.currentBackStackEntryAsState
+import com.dcs.presentation.main.Screen.Main
+
+@Composable
+fun MainBottomNavigation(
+    items: List<Main> =
+        listOf(
+            Main.Home,
+            Main.Search,
+            Main.People,
+            Main.Setting,
+        ),
+    navController: NavHostController,
+    modifier: Modifier = Modifier,
+) {
+    val navBackStackEntry by navController.currentBackStackEntryAsState()
+    val currentRoute = navBackStackEntry?.destination?.route
+
+    AnimatedVisibility(visible = items.any { it.route == currentRoute }) {
+        NavigationBar(
+            modifier = modifier,
+        ) {
+            items.forEach { screen ->
+                NavigationBarItem(
+                    label = {
+                        Text(text = stringResource(id = screen.titleResId))
+                    },
+                    selected = currentRoute == screen.route,
```

Comment:
네비게이션 탭의 계층 구조 상에서 현재 선택된 화면의 선택 여부를 처리할려면 단순하게 destination의 route를 비교하는 것보다 hierarchy에 있는 route를 비교하는 것이 좋을 것 같음.

Reply:
```suggestion
                    selected = navBackStackEntry?.destination?.hierarchy?.any { it.route == screen.route } == true
```
any를 통해서 추가 비교 연산도 생기고 코드가 다소 지저분해지기는 한데요...
형이 말씀하신 장점은 어떤게 있을까요?

Reply:
메인 탭 네개를 제외한 Route 가 호출 될 경우 최상위를 destination으로 구분 못한다는 걸로 알고 있는데 확인 바람.

Reply:
수정 완료
https://github.com/Edgar-J-Shin/AllAboutMovie/pull/3/commits/c614c75118f7bb560477c55e610c52510d52e319